lib/gateway/Shard.js

```diff
@@ -11,31 +11,17 @@ const {GATEWAY_VERSION, GatewayOPCodes, ChannelTypes} = require("../Constants");
 const ExtendedUser = require("../structures/ExtendedUser");
 const User = require("../structures/User");
 let WebSocket = typeof window !== "undefined" ? window.WebSocket : require("ws");
+const {safeRequire} = require("../util/utils");
+WebSocket = safeRequire("uws");
```

Comment:
This will cause Websocket to be undefined if `uws` is not installed instead of using the default "ws" as it used to.
```suggestion
WebSocket = safeRequire("uws") || WebSocket;
```